clients/src/test/java/org/apache/kafka/clients/consumer/CooperativeStickyAssignorTest.java

```diff
@@ -78,6 +80,26 @@ public void testDecodeGeneration() {
         assertFalse(((CooperativeStickyAssignor) assignor).memberData(subscription).generation.isPresent());
     }
 
+    @Test
```

Comment:
All the tests use `buildSubscriptionWithGeneration` and `buildSubscription` which rely on the old method to encode the partitions and the generation. I wonder if we should update those in order to use the latest mechanism. What do you think? Then, we should have a few tests to ensure the backward compatibility.

Reply:
I agree we should add tests for the latest mechanism, but I think we should still keep the tests for old schema. These are unit tests, which will not take too long. I added variable to test old and new schemas.

Reply:
I am not entirely convinced by adding version to all tests because it makes the tests harder to read. I would rather used the most up to date scheme in the tests and add new unit test for the backward compatibility. This seems more explicit and would keep existing test simpler to read.

Reply:
OK, tests updated. For existing tests, we'll use the latest scheme to build subscription. And added some backward compatibility tests.